docs/Installation-Guide.rst

```diff
@@ -84,26 +84,71 @@ LightGBM uses **CMake** to build. Run the following commands:
 
 Also you may want to reed `gcc Tips <./gcc-Tips.rst>`__.
 
-OSX
-~~~
+macOS
+~~~~~
 
-LightGBM depends on **OpenMP** for compiling, which isn't supported by Apple Clang.
+LightGBM depends on **OpenMP** for compiling, which isn't supported by default Apple Clang.
 
-Please install **gcc/g++** by using the following commands:
+Make sure you have Apple's **Command Line Tools** installed
 
 .. code::
 
-  brew install cmake
-  brew install gcc --without-multilib
+   gcc
+   xcode-select --install
+
+Although the prompt message is a bit confusing, clicking **install** and
+following the instructions will install **Command Line Tools**, not *Xcode*
+
+
+Install GNU Compiler Collection, **gcc with OpenMP enabled**
 
-Then install LightGBM:
+Choose the binaries version number based on your OS `(for more info) <http://hpc.sourceforge.net>`_.
 
 .. code::
+   
+   High Sierra, Sierra & El Capitan : gcc-7.1-bin.tar.gz
+   Yosemite: gcc-5.1-bin.tar.gz
 
-  git clone --recursive https://github.com/Microsoft/LightGBM ; cd LightGBM
-  export CXX=g++-7 CC=gcc-7
-  mkdir build ; cd build
-  cmake ..
+.. code::
+   
+   cd Downloads
+   wget http://prdownloads.sourceforge.net/hpc/gcc-7.1-bin.tar.gz
+   # This will install the compiler in `/usr/local/bin`
+   sudo tar -zxvf gcc-7.1-bin.tar.gz -C/
+   # to remove the package from your Downloads directory
+   sudo rm gcc-7.1-bin.tar.gz
+   # to check the installation
+   gcc -v
+
+Install *`CMake <https://cmake.org/overview/>`_*
+
+.. code-block:: bash
+
+   cd Downloads
+   wget https://cmake.org/files/v3.10/cmake-3.10.0-rc3-Darwin-x86_64.dmg
+   # double-click on the disk image and follow instructions
+   
+To add CMake to the PATH:
+
+.. code-block:: bash
+   
+    PATH="/Applications/CMake.app/Contents/bin":"$PATH"
+    # Or, to install symlinks to '/usr/local/bin', run:
+    sudo "/Applications/CMake.app/Contents/bin/cmake-gui" --install
+    # Or, to install symlinks to another directory, run:
+    sudo "/Applications/CMake.app/Contents/bin/cmake-gui" --install=/path/to/bin
+
+
+Then, run the following commands to install LightGBM:
+
+.. code::
+
+  cd Downloads
+  git clone --recursive https://github.com/Microsoft/LightGBM
+  cd LightGBM; mkdir build ; cd build
+  export CC=/usr/local/bin/gcc CXX=/usr/local/bin/g++ CXX1X=/usr/local/bin/g++ # with GCC
+  cmake ../CMakeLists.txt
+  cd ..
```

Comment:
Why do we need this (and don't need in the MPI case?)?

Reply:
Did not test with MPI

Reply:
I was not able to have a successful build with MPI as stated above, but yes it's needed too.

Reply:
We currently have the following working piece of code:
https://github.com/Microsoft/LightGBM/blob/66c81af2a62fa367063c3d180e58087ab3eeb4ae/.travis/test.sh#L97-L106

And there is no `cd ..` here.

Reply:
We are not talking about the same thing.
Have you at least tried it ?

Reply:
No, I didn't try your code.

Reply:
@Laurae2 and @StrikerRUS is right. You don't need cmake ../CMakeLists.txt, you are installing all the things into LightGBM folder, not build folder. You don't even need mkdir folder in this case.

Our current installation steps work both on my macOS and our travis test machine.